packages/event-listener/src/component.ts

```diff
@@ -0,0 +1,26 @@
+import { isClient } from "@solid-primitives/utils";
+import { Component, JSX } from "solid-js";
+import { createEventListener } from ".";
+
+type EventAttributes<T> = {
+  [K in keyof JSX.DOMAttributes<T>]: `${K}` extends `on${string}` ? JSX.DOMAttributes<T>[K] : never;
+};
+
+/**
+ * Listen to the `window` DOM Events, using a component.
+ *
+ * @see https://github.com/davedbase/solid-primitives/tree/main/packages/event-listener#GlobalEventListener
+ *
+ * @example
+ * <GlobalEventListener onMouseMove={e => console.log(e.x, e.y)} />
+ */
+export const GlobalEventListener: Component<EventAttributes<null>> = props => {
+  if (isClient) {
+    Object.keys(props).forEach(attr => {
+      if (!attr.startsWith("on")) return;
```

Comment:
This is interesting. I guess I wanted to limit it to only events by design because it is a "GlobalEventListener".
If it would accept other attributes as well that to what they should be attached to? The body tag, HTML tag? Because I don't think they can be attached to the window object.
It is an interesting idea, but maybe making a separate component would be better. And keep GlobalEventListener only for window/document events.

Reply:
You could rename it to `WindowEventListener` and `DocumentEventListener`.